### PR TITLE
Project 3 tutorial steps 1-3

### PR DESCRIPTION
**Pull request by kwell755**
This is Steps 1-3 for the week 3 server tutorial. This includes the servlets, fetch, and JSON walkthroughs. Will spit each step into different PR moving forward.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,6 +27,6 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
```
Comment (samouri) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: Its okay to remove the `<h1>`, but you don't need to if you don't want to! The [h1](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements) tag  signifies that the text is a "header" text.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -40,4 +40,33 @@ function KaylaTrivia(){
     else{
          alert(`Your final score is ${points}  Oof maybe try again?`);
     }
+}
+function getMessage() {
+  console.log('Fetching a random quote.');
```
Comment (caroqliu) [inline portfolio/src/main/webapp/script.js:45]: While printing to the console is useful for development/debugging, these lines generally don't make it into the code. Feel free to use locally but I'd recommend not committing it with the code (here and elsewhere below).

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -40,4 +40,33 @@ function KaylaTrivia(){
...
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addQuoteToDom);
+}
+
+/** Adds a random quote to the DOM. */
```
Comment (caroqliu) [inline portfolio/src/main/webapp/script.js:66]: Nice job leaving code comments! Small recommendation: Try adding to your jsdoc to include the `@param` annotation: https://google.github.io/styleguide/jsguide.html#jsdoc-general-form

Comment (caroqliu) [reply portfolio/src/main/webapp/script.js:66]: Also feel free to update the description. It doesn't look like it adds a random quote to the DOM -- it adds the quote that the function was given as an argument to the DOM :)

**Comment by samouri** [comment]
> It looks like your PR is still in "Draft mode" even though you've added reviewers. Feel free to click "Ready for Review" to change the PR from "Draft" to "Open".


@caroqliu: My b I added the reviewers!

**Pull request merged**